src/mono/mono/mini/method-to-ir.c

```diff
@@ -7543,8 +7544,8 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 					MonoType *ptype = fsig->params [idx_param];
 					MonoClass *klass = mono_class_from_mono_type_internal (ptype);
 
-					// SwiftSelf and SwiftError are special cases where we need to preserve the class information for the codegen to handle them correctly.
-					if (mono_type_is_struct (ptype) && !(klass == swift_self || klass == swift_error)) {
+					// SwiftSelf, SwiftError, and SwiftIndirectResult are special cases where we need to preserve the class information for the codegen to handle them correctly.
+					if (mono_type_is_struct (ptype) && !(klass == swift_self || klass == swift_error || klass == swift_indirect_result)) {
```

Comment:
Nit. in the future we might start thinking about replacing the `!(klass == swift_self || klass == swift_error || klass == swift_indirect_result)` with a function, e.g. `is_swift_codegen_class (klass)` (probably better name needed). Especially, if the `SwiftSelf<T>` requires additional case here and in `transform.c`.

Reply:
Agree. Let's handle it in follow-up PRs as more changes may be performed.